### DISCUSSION TITLE
Single project support without MAUI dependency

### DISCUSSION THREAD

**Discussion by prollin** [dotnet/maui#10986] [Ideas]
I have a fairly large application with separate project for net6-windows, net6-ios and net6-android.
Is there a way to have a single project that can deploy (through VS Windows) to each of those platform *without* MAUI?

Currently, a project that targets both net6-ios and net6-android seem to not show any way to run on device from VS user interface (unless it is a MAUI App). It seems that this should be a feature of net6, not MAUI.

**Comment by xtuzy** [comment]
i remember have a singleproject tag in .csproj file, maybe you can try.
i used maui tag in https://github.com/xtuzy/MauiAppTemplate, i feel you can remove maui dll when Linker

**Comment by yunusefendi52** [comment]
Single project **_is_** actually a concept of dotnet (not maui), but I haven't found any documentation yet.

I think you can use `SingleProject` property in csproj
```csproj
<PropertyGroup>
<SingleProject>true</SingleProject> <!-- add this -->
</PropertyGroup>
```

See also how uno platform does this on their template https://github.com/unoplatform/uno/blob/master/src/SolutionTemplate/UnoSolutionTemplate.WinUI.netcore/Mobile/UnoQuickStart.Mobile.csproj

**Comment by prollin** [comment]
> Single project is actually a concept of dotnet (not maui), but I haven't found any documentation yet.

All the documentation about this feature seem to be tied to MAUI; 
see https://github.com/dotnet/maui/wiki/Single-Project or https://learn.microsoft.com/en-us/dotnet/maui/fundamentals/single-project

Regardless of this, Visual Studio UI to deploy to device doesn't appear when not using MAUI.

Here is what I would expect:
- Put all platform specific code/resources in `Platforms` folder (`AppDelegate`/`MainActivity`)
- VS detects the multitargeting and offers the user options to deploy on each connected device (as it does with a default MAUI app)
- Compiler automatically skip code/resources that are not for the device the app is being executed on

Are there any workaround to achieve this?
Note that the reason for not using MAUI is because the app in question is a game that manages its own view/surface and perform its own rendering.

**Comment by prollin** [comment]
> i used maui tag in https://github.com/xtuzy/MauiAppTemplate, i feel you can remove MAUI dll when Linker

@xtuzy I seem to be able to see the deployment UI in Visual studio when using the `<UseMaui>` tag along with `<SingleProject>` as your example shows.

Using `<UseMauiCore>` doesn't work though (Getting `Specified AndroidManifest file does not exist` compilation error).

Is there any plan to make this behavior work without the need for `<UseMaui>` tag at all?

**Reply by xtuzy** [reply]
Sorry, i try to not use Maui tag, but fail. Advice you to ask maui team(they don't read Discussions usually).

**Reply by yunusefendi52** [reply]
or maybe also ask uno platform team how they do it? I only tried it in uno platform (their template) and not creating manually
